src/core_plugins/kibana/public/visualize/editor/agg_params.js

```diff
@@ -81,15 +81,18 @@ uiModules
         // build collection of agg params html
         type.params.forEach(function (param, i) {
           let aggParam;
+          let fields;
           // if field param exists, compute allowed fields
           if (param.name === 'field') {
-            $aggParamEditorsScope.indexedFields = getIndexedFields(param);
+            fields = $aggParamEditorsScope.indexedFields = getIndexedFields(param);
+          } else if (param.type === 'field') {
+            fields = $aggParamEditorsScope[param.name] = getIndexedFields(param);
```

Comment:
Ah good point, I didn't realize all aggParam editors share the same scope. How about appending a suffix to the variable name to make it more obvious in the template what this variable holds? So instead of `$aggParamEditorsScope[param.name]` we could do something like 
```
$aggParamEditorsScope[`${param.name}Options`]
```